relay-server/src/actors/project.rs

```diff
@@ -623,23 +641,15 @@ impl Project {
             }
         }
 
-        let state = self.state.as_ref();
-        let expiry = state
-            .map(|s| s.check_expiry(&self.config))
-            .unwrap_or(Expiry::Expired);
-
-        let cached_state = match (state, expiry) {
+        let cached_state = match self.expiry_state() {
```

Comment:
Please check this new match statement carefully. This should exactly match the old logic.

Reply:
state None corresponds to ExpiryState expired, so I think we are good.